soh/soh/Enhancements/randomizer/3drando/shops.cpp

```diff
@@ -31,82 +31,85 @@ void PlaceVanillaShopItems() {
 //Shopsanity random will have anywhere from the first 32 to the first 56, so the order of items after 32 is relevant
 std::vector<RandomizerGet> GetMinVanillaShopItems(int total_replaced) {
     std::vector<RandomizerGet> minShopItems = {
-    RG_BUY_DEKU_SHIELD,
-    RG_BUY_HYLIAN_SHIELD,
-    RG_BUY_GORON_TUNIC,
-    RG_BUY_ZORA_TUNIC,
-    RG_BUY_DEKU_NUTS_5,
-    RG_BUY_DEKU_NUTS_5,
-    RG_BUY_DEKU_NUTS_10,
-    RG_BUY_DEKU_STICK_1,
-    RG_BUY_DEKU_STICK_1,
-    RG_BUY_DEKU_SEEDS_30,
-    RG_BUY_ARROWS_10,
-    RG_BUY_ARROWS_10,
-    RG_BUY_ARROWS_30,
-    RG_BUY_ARROWS_50,
-    RG_BUY_BOMBCHUS_10,
-    RG_BUY_BOMBCHUS_10,
-    RG_BUY_BOMBCHUS_10,
-    RG_BUY_BOMBCHUS_20,
-    RG_BUY_BOMBS_525,
-    RG_BUY_BOMBS_535,
-    RG_BUY_BOMBS_10,
-    RG_BUY_BOMBS_20,
-    RG_BUY_GREEN_POTION,
-    RG_BUY_RED_POTION_30,
-    RG_BUY_BLUE_FIRE,
-    RG_BUY_FAIRYS_SPIRIT,
-    RG_BUY_BOTTLE_BUG,
-    RG_BUY_FISH,
-    //^First 28 items from OoTR
-    RG_BUY_HYLIAN_SHIELD,
-    RG_BUY_BOTTLE_BUG,
-    RG_BUY_DEKU_STICK_1,
-    RG_BUY_FAIRYS_SPIRIT,
-    //^First 32 items: Always guaranteed
-    RG_BUY_BLUE_FIRE,
-    RG_BUY_FISH,
-    RG_BUY_BOMBCHUS_10,
-    RG_BUY_DEKU_NUTS_5,
-    RG_BUY_ARROWS_10,
-    RG_BUY_BOMBCHUS_20,
-    RG_BUY_BOMBS_535,
-    RG_BUY_RED_POTION_30,
-    //^First 40 items: Exist on shopsanity 3 or less
-    RG_BUY_BOMBS_30,
-    RG_BUY_BOMBCHUS_20,
-    RG_BUY_DEKU_NUTS_5,
-    RG_BUY_ARROWS_10,
-    RG_BUY_DEKU_NUTS_5,
-    RG_BUY_ARROWS_30,
-    RG_BUY_RED_POTION_40,
-    RG_BUY_FISH,
-    //^First 48 items: Exist on shopsanity 2 or less
-    RG_BUY_BOMBCHUS_20,
-    RG_BUY_ARROWS_30,
-    RG_BUY_RED_POTION_50,
-    RG_BUY_ARROWS_30,
-    RG_BUY_DEKU_NUTS_5,
-    RG_BUY_ARROWS_50,
-    RG_BUY_ARROWS_50,
-    RG_BUY_GREEN_POTION,
-    //^First 56 items: Exist on shopsanity 1 or less
-    RG_BUY_POE,
-    RG_BUY_POE,
-    RG_BUY_HEART,
-    RG_BUY_HEART,
-    RG_BUY_HEART,
-    RG_BUY_HEART,
-    RG_BUY_HEART,
-    RG_BUY_HEART,
-    //^All 64 items: Only exist with shopsanity 0
-  };
-  //Now delete however many items there are to replace
-  for (int i = 0; i < total_replaced; i++) {
-    minShopItems.pop_back();
-  }
-  return minShopItems;
+        RG_BUY_DEKU_SHIELD,
+        RG_BUY_HYLIAN_SHIELD,
+        RG_BUY_GORON_TUNIC,
+        RG_BUY_ZORA_TUNIC,
+        RG_BUY_DEKU_NUTS_5,
+        RG_BUY_BOMBS_20,
+        RG_BUY_BOMBCHUS_10,
+        RG_BUY_DEKU_STICK_1,
+        //^First 8 items: Exist on shopsanity 7 or less
+        RG_BUY_DEKU_STICK_1,
```

Comment:
There should be much more variety in the second section then a bunch of arrows: Having Green potion is the only non-drop, non-granny source of magic refills. Having a Red Potion and Heart for no-drops healing would be nice. Blue fire because buy blue fire is something that can come up in logic. Fairies can matter once fairy shuffle is in.  After all that, a variety of ammo would make shops much more useful in general.

Reply:
Added all except heart since there was none in the 28 items from n64 rando.